src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala

```diff
@@ -190,7 +190,13 @@ class ErgoWalletActor(settings: ErgoSettings,
           historyReader.bestFullBlockAt(blockHeight) match {
             case Some(block) =>
               log.info(s"Wallet is going to scan a block ${block.id} in the past at height ${block.height}")
-              ergoWalletService.scanBlockUpdate(state, block)
+              ergoWalletService.scanBlockUpdate(state, block) match {
+                case Failure(ex) =>
+                  log.error("Scanning past block update failed", ex)
+                  state
```

Comment:
@kushti I'm wondering whether we should stop scanning in the past if this happens or continue :shrug: We stopped here before this pull request in case of a failure.

Reply:
@pragmaxim I guess if we stop, there should be some new wallet status introduced (Failed) and the status should be passed into the UI. Sounds like a topic for another issue